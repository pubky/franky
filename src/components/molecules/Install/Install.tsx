'use client';

import Image from 'next/image';
import { useRouter } from 'next/navigation';
import { AppWindow, ArrowRight } from 'lucide-react';

import * as Atoms from '@/atoms';
import * as Molecules from '@/molecules';
import * as Libs from '@/libs';
import { Button } from '@/atoms';

export const InstallCard = () => {
  return (
    <Molecules.ContentCard
      image={{
        src: '/images/keyring.png',
        alt: 'Keyring',
        width: 192,
        height: 192,
        size: 'large',
      }}
    >
      <Atoms.Container>
        <Image src="/images/logo-pubky-ring.svg" alt="Pubky Ring" width={220} height={48} />
        <Atoms.Typography className="text-secondary-foreground opacity-80 font-normal">
          Download and install the mobile app to start creating your account.
        </Atoms.Typography>
      </Atoms.Container>
      <StoreButtons />
    </Molecules.ContentCard>
  );
};

export const InstallFooter = () => {
  return (
    <Atoms.FooterLinks className="mt-6">
      Use{' '}
      <Atoms.Link href="https://www.pubkyring.to/" target="_blank">
        Pubky Ring
      </Atoms.Link>{' '}
      or any other{' '}
      <Atoms.Link href="https://pubky.org" target="_blank">
        Pubky Core
      </Atoms.Link>
      â€“powered keychain, or create your keys in the browser (less secure).
    </Atoms.FooterLinks>
  );
};

export const InstallHeader = () => {
  return (
    <Atoms.PageHeader>
      <Molecules.PageTitle size="large">
        Install <span className="text-brand">Pubky Ring.</span>
      </Molecules.PageTitle>
      <Atoms.PageSubtitle>Pubky Ring is a keychain for your identity keys in the Pubky ecosystem.</Atoms.PageSubtitle>
    </Atoms.PageHeader>
  );
};

export const InstallNavigation = ({ ...props }: React.HTMLAttributes<HTMLDivElement>) => {
  const router = useRouter();

  const handleCreate = () => {
    router.push('/onboarding/pubky');
  };

  const handleContinue = () => {
    router.push('/onboarding/scan');
  };

  return (
    <Atoms.Container className={Libs.cn('flex-col-reverse lg:flex-row gap-3 lg:gap-6', props.className)}>
      <Atoms.Container className="items-center gap-1 flex-row">
        <Atoms.Button variant="outline" className="rounded-full flex-1 md:flex-none" onClick={handleCreate}>
          <AppWindow className="mr-2 h-4 w-4" />
          Create keys in browser
        </Atoms.Button>
        <Molecules.PopoverTradeoffs />
      </Atoms.Container>
      <Button size="lg" className="rounded-full" onClick={handleContinue}>
        <ArrowRight className="mr-2 h-4 w-4" />
        Continue with Pubky Ring
      </Button>
    </Atoms.Container>
  );
};

export function StoreButtons({ className }: React.HTMLAttributes<HTMLDivElement>) {
  return (
    <Atoms.Container className={Libs.cn('flex-row gap-4 justify-around sm:justify-start', className)}>
      <Molecules.DialogDownloadPubkyRing store="apple" />
      <Molecules.DialogDownloadPubkyRing store="android" />
    </Atoms.Container>
  );
}
