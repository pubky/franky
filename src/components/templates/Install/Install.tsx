import * as Molecules from '@/molecules';

export function Install() {
  return (
    <Molecules.PageContainer>
      <Molecules.InstallHeader />
      <Molecules.InstallCard />
      <Molecules.InstallFooter />
      <Molecules.InstallNavigation />
    </Molecules.PageContainer>
  );
}
