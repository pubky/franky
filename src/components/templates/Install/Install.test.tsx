import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Install } from './Install';

// Mock atoms
vi.mock('@/atoms', () => ({
  Container: ({ children, size }: { children: React.ReactNode; size?: string }) => (
    <div data-testid="container" className={`container ${size || ''}`}>
      {children}
    </div>
  ),
}));

// Mock molecules
vi.mock('@/molecules', () => ({
  PageContainer: ({ children }: { children: React.ReactNode }) => <div data-testid="page-container">{children}</div>,
  InstallHeader: () => <div data-testid="install-header">Install Header</div>,
  InstallCard: () => <div data-testid="install-card">Install Card</div>,
  InstallFooter: () => <div data-testid="install-footer">Install Footer</div>,
  InstallNavigation: () => <div data-testid="install-navigation">Install Navigation</div>,
}));

describe('Install', () => {
  it('renders all main components', () => {
    render(<Install />);

    expect(screen.getByTestId('container')).toBeInTheDocument();
    expect(screen.getByTestId('install-header')).toBeInTheDocument();
    expect(screen.getByTestId('install-card')).toBeInTheDocument();
    expect(screen.getByTestId('install-footer')).toBeInTheDocument();
    expect(screen.getByTestId('install-navigation')).toBeInTheDocument();
  });

  it('renders components in correct order within page container', () => {
    render(<Install />);

    const pageContainer = screen.getByTestId('container');
    const children = Array.from(pageContainer.children);

    expect(children).toHaveLength(4);
    expect(children[0]).toHaveAttribute('data-testid', 'install-header');
    expect(children[1]).toHaveAttribute('data-testid', 'install-card');
    expect(children[2]).toHaveAttribute('data-testid', 'install-footer');
    expect(children[3]).toHaveAttribute('data-testid', 'install-navigation');
  });

  it('wraps all content in page container', () => {
    render(<Install />);

    const pageContainer = screen.getByTestId('container');

    // All main components should be children of PageContainer
    expect(pageContainer).toContainElement(screen.getByTestId('install-header'));
    expect(pageContainer).toContainElement(screen.getByTestId('install-card'));
    expect(pageContainer).toContainElement(screen.getByTestId('install-footer'));
    expect(pageContainer).toContainElement(screen.getByTestId('install-navigation'));
  });

  it('renders without crashing', () => {
    const { container } = render(<Install />);
    expect(container.firstChild).toBeInTheDocument();
  });

  it('has correct component hierarchy', () => {
    render(<Install />);

    // Verify the main container exists
    const pageContainer = screen.getByTestId('container');
    expect(pageContainer).toBeInTheDocument();

    // Verify all expected child components exist
    const expectedComponents = ['install-header', 'install-card', 'install-footer', 'install-navigation'];

    expectedComponents.forEach((componentTestId) => {
      expect(screen.getByTestId(componentTestId)).toBeInTheDocument();
    });
  });
});
