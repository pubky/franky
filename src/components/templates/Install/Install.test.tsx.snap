// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Install - Snapshots > matches snapshot 1`] = `
<div
  class="mx-auto w-full flex-col flex container min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
  data-testid="container"
>
  <div
    class="flex w-full flex-1 flex-col gap-0 lg:flex-none"
    data-testid="install-content"
  >
    <div
      data-testid="install-header"
    >
      Install Header
    </div>
    <div
      data-testid="install-card"
    >
      Install Card
    </div>
    <div
      data-testid="install-footer"
    >
      Install Footer
    </div>
  </div>
  <div
    class="onboarding-nav mt-auto w-full lg:mt-0"
  >
    <div
      data-testid="install-navigation"
    >
      Install Navigation
    </div>
  </div>
</div>
`;
